include/multipass/utils.h

```diff
@@ -97,6 +98,18 @@ void validate_server_address(const std::string& value);
 bool valid_hostname(const std::string& name_string);
 std::string generate_mac_address();
 bool valid_mac_address(const std::string& mac);
+template <typename NetworkContainer>
+std::optional<NetworkInterfaceInfo> find_bridge_with(const NetworkContainer& networks,
+                                                     const std::string& member_network,
+                                                     const std::string& bridge_type)
+{
```

Comment:
About the 3rd parameter `bridge_type`, since the `MP_PLATFORM.bridge_nomenclature()` is a singleton class member function. Can we remove this parameter and retrieve the bridge_type inside the function?

Reply:
Also, callers already need to interrogate the platform for this and other stuff. For example: https://github.com/canonical/multipass/blob/ac1629cff7c677a1573f557b8b7844d2825ca3e0/src/platform/backends/shared/base_virtual_machine_factory.cpp#L71 and https://github.com/canonical/multipass/blob/46afc1cc266e33e20b13cb163eaef98fd7da85d4/src/platform/backends/lxd/lxd_virtual_machine_factory.cpp#L236

Don't you think it is better for "utilities" to be independent of the platform? Let me know your thoughts.

Reply:
Hmm, indeed. making `utilities` depends on `MP_PLATFORM` may not be ideal. It looks like all callers are either `virtual_machine_factory` or `daemon`. Then moving the `find_bridge_with` function from `utilities` to `base_virtual_machine_factory.h` and make it a free function there might be a better choice. However, this requires more code change, I think we can leave it for now.

Reply:
OK, thanks. I reconsidered moving it there, but notice that this is totally independent from the factory or the backends. It really is a generic utility: just a find over a container for a certain network. We could have it as a static in the factory, but that would kind of break single-responsibility, locality...

Reply:
> We could have it as a static in the factory,

Do you mean `BaseVirtualMachineFactory::find_bridge_with` static function? That is similar with a free function in `base_virtual_machine_factory.h` but with access to data member of the BaseVirtualMachineFactory which is not needed.

Reply:
Yep, the static function only accesses static data members. I have mixed that with friend function. As to free function might pollute file `base_virtual_machine_factory.h`, indeed, this utility is not really coupled with the `base_virtual_machine_factory` class. Overral, I think we can leave it for now.